pkg/provider/github/github.go

```diff
@@ -543,3 +543,33 @@ func uniqueRepositoryID(repoIDs []int64, id int64) []int64 {
 	}
 	return r
 }
+
+// listBranches list all branches in the repo and set the default branch for a matching branch commit SHA.
+func (v *Provider) listBranches(ctx context.Context, runevent *info.Event) error {
+	if v.Client == nil {
+		return fmt.Errorf("no github client has been initialized, " +
+			"exiting... (hint: did you forget setting a secret on your repo?)")
+	}
+
+	opt := github.ListOptions{PerPage: v.paginedNumber}
+	for {
+		branchInfo, resp, err := v.Client.Repositories.ListBranches(ctx, runevent.Organization, runevent.Repository,
+			&github.BranchListOptions{
+				ListOptions: opt,
+			})
+		if err != nil {
+			return err
+		}
+		for _, b := range branchInfo {
+			if b.Commit.GetSHA() == runevent.SHA {
```

Comment:
This only get for the first commit of the branch? this is intended, right?

Reply:
usually yes because
![Screenshot from 2023-10-24 19-09-32](https://github.com/openshift-pipelines/pipelines-as-code/assets/9441662/2b2158a7-3940-4125-b6d5-6fc668b0a161)
only latest commit will have main branch and remaining commits will have different branch from which PR raised

Branch can be found using
```
git branch --contains 4502b19559c0d5b456bd35b18a6f9b6615d73898
```

Reply:
i am not sure i undestand what about if the retest is done on another branch on the repo that is not on the  main branch?

Reply:
i am asking about this part 

				runevent.DefaultBranch = b.GetName()

Reply:
Yes if push request is done on other than main branch then defaultbranch will be that branch



I verified here
https://github.com/savitaashture/articleyamls/commit/a90c71bcf530ded4b3f615480a9fba566ec809bd

Reply:
I am afraid rewriting the defaultBranch setting to something that is fake may bring some bad side effect, for example on fetching the policies is it necessary?

Reply:
what about if the commit is not at the top of the branch and user does a /retest

your code checks only the commit that is first in the branch not the ones after, what happen here? what should happen if that doesn't work?

```go

		for _, b := range branchInfo {
			if b.Commit.GetSHA() == runevent.SHA {
```